File: backend-apps/backend/src/main/java/com/shaderock/lunch/backend/feature/config/preference/company/entity/CompanyPreferences.java
package com.shaderock.lunch.backend.feature.config.preference.company.entity;

import com.shaderock.lunch.backend.data.entity.DeletableEntity;
import com.shaderock.lunch.backend.feature.company.entity.Company;
import com.shaderock.lunch.backend.feature.config.preference.company.type.CompanyDiscountType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import java.time.LocalTime;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.ToString.Exclude;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
@ToString
@Entity
public class CompanyPreferences extends DeletableEntity {

  @OneToOne(fetch = FetchType.LAZY, optional = false)
  @JoinColumn(name = "company_id", nullable = false)
  @Exclude
  private Company company;

  @Column
  @Enumerated(EnumType.STRING)
  private CompanyDiscountType companyDiscountType;

  @Column
  private Integer discountPercentageFirstOrder;

  @Column
  private Double discountFixFirstOrder;

  @Column
  private Double maxDiscountFixFirstOrder;

  @Column
  private Double discountPerDay;

  @Column
  private String deliveryAddress;

  @Column
  private LocalTime deliverAt;

  @Builder(builderMethodName = "baseEntityBuilder")
  public CompanyPreferences(UUID id, boolean isDeleted) {
    super(id, isDeleted);
  }
}

File: frontend/src/models/CompanyPreferences.ts
import {CompanyDiscountType} from "@/models/CompanyDiscountType";

export class CompanyPreferences {
  public id: string | null
  public companyId: string | null
  public companyDiscountType: CompanyDiscountType
  public discountPercentageFirstOrder: number
  public discountFixFirstOrder: number
  public maxDiscountFixFirstOrder: number
  public discountPerDay: number
  public deliverAt: Date
  public deliveryAddress: string

  constructor(id: string | null,
              companyId: string | null,
              companyDiscountType:
                CompanyDiscountType,
              discountPercentageFirstOrder: number,
              discountFixFirstOrder: number,
              maxDiscountFixFirstOrder: number,
              discountPerDay: number,
              deliverAt: Date,
              deliveryAddress: string) {
    this.id = id;
    this.companyId = companyId;
    this.companyDiscountType = companyDiscountType;
    this.discountPercentageFirstOrder = discountPercentageFirstOrder;
    this.discountFixFirstOrder = discountFixFirstOrder;
    this.maxDiscountFixFirstOrder = maxDiscountFixFirstOrder;
    this.discountPerDay = discountPerDay;
    this.deliverAt = deliverAt;
    this.deliveryAddress = deliveryAddress;
  }
}
